labs/setup-pattern.md

```diff
@@ -0,0 +1,72 @@
+# The Setup Pattern
+
+One pattern that bot developers might find useful when using the Matrix Rust SDK is to use a "setup"
+pattern, which, simply put, consists of having a bot user bootstrap the bot through a dedicated
+`setup` (sub)command, and then run the main executable.
+
+Then, together with that, create an invariant in the main executable: Always expect state.
+
+The latter can be done by checking if the path to a stored and serialized
+[`Session`](https://docs.rs/matrix-sdk/latest/matrix_sdk/struct.Session.html) configuration exists
+(as created during the `setup` command), and/or by checking for files under the path provided to
+[`make_store_config`](https://docs.rs/matrix-sdk/latest/matrix_sdk/store/fn.make_store_config.html).
+
+If the main executable can't find this configuration, abort running.
+
+## Context
+
+The context of this pattern document, and the reason for the crystallization of this idea, is that
+currently the "modus operandi" of Instant Messaging bots is quite stateless. Bots might still have
+state pertaining to users, databases and whatnot that contain explicit state of business logic, but
+it's not expected that the *bot itself* has state.
```

Comment:
I don't find the "bot itself" to be a clear concept. Maybe word this as "it's not expected that even a simple hello-world bot has state" (or "needs state")?

Reply:
Application state, business logic state, sorry, I could've worded this better like that.